src/Lorentz/Contracts/BaseDAO/TZIP16Metadata.hs

```diff
@@ -106,6 +108,39 @@ isOperatorView = View
       ]
   }
 
+tokenMetadataView :: DaoView
+tokenMetadataView = View
+  { vName = "token_metadata"
+  , vDescription = Just
+      "Returns metadata for given token according to TZIP-12."
+  , vPure = Just True
+  , vImplementations =
+      [ VIMichelsonStorageView $
+          mkMichelsonStorageView
+           @MetadataStorage
+           @(FA2.TokenId, FA2.TokenMetadata)
+           Nothing [] $
+            WithParam @FA2.TokenId $ do
+              dip $ do
+                drop @MetadataStorage
+                push $ Map.fromList
+                  [ (frozenTokenId, cFrozenTokenMetadata config)
+                  , (unfrozenTokenId, cUnfrozenTokenMetadata config)
+                  ]
+              dup
+              dip $ do
+                get
+                ifSome nop $ failCustom_ #fA2_TOKEN_UNDEFINED
+              pair
+      ]
+  }
+  where
+    -- TODO: what to do here?
+    -- Accept token metadatas in CLI?
```

Comment:
I think ideally `print-metadata` should have options like `--token-name` with some sane default values.